src/source/src/lib.rs

```diff
@@ -66,13 +70,64 @@ pub enum SourceImpl {
     Connector(ConnectorSource),
 }
 
+pub enum SourceStreamReaderImpl {
+    TableV2(TableV2StreamReader),
+    Connector(ConnectorStreamReader),
+}
+
+#[async_trait]
+impl StreamSourceReader for SourceStreamReaderImpl {
+    async fn open(&mut self) -> Result<()> {
+        match self {
+            SourceStreamReaderImpl::TableV2(t) => t.open().await,
+            SourceStreamReaderImpl::Connector(c) => c.open().await,
+        }
+    }
+
+    async fn next(&mut self) -> Result<StreamChunk> {
+        match self {
+            SourceStreamReaderImpl::TableV2(t) => t.next().await,
+            SourceStreamReaderImpl::Connector(c) => c.next().await,
+        }
+    }
+}
+
+pub enum SourceReaderContext {
```

Comment:
```suggestion
pub type SourceReaderContext = Vec<SplitImpl>;
```

None can be represented as empty vec. And what is the difference with `pub(crate) struct ConnectorReaderContext`?

Reply:
This is a semantic difference; table sources don't need splits to be initialized, so `None` is more appropriate than an empty Vec